db/db_impl/db_impl.cc

```diff
@@ -3401,16 +3401,33 @@ Status DBImpl::GetApproximateSizes(const SizeApproximationOptions& options,
     return Status::InvalidArgument("Invalid options");
   }
 
+  const Comparator* const ucmp = column_family->GetComparator();
+  assert(ucmp);
+  size_t ts_sz = ucmp->timestamp_size();
+
   Version* v;
   auto cfh = static_cast_with_check<ColumnFamilyHandleImpl>(column_family);
   auto cfd = cfh->cfd();
   SuperVersion* sv = GetAndRefSuperVersion(cfd);
   v = sv->current;
 
   for (int i = 0; i < n; i++) {
+    Slice start = range[i].start;
+    Slice limit = range[i].limit;
+
+    // Add timestamp if needed
+    std::string start_with_ts, limit_with_ts;
+    if (ts_sz > 0) {
+      // Append a minimal timestamp as the range limit is exclusive:
+      // [start, limit)
+      AppendKeyWithMinTimestamp(&start_with_ts, start, ts_sz);
```

Comment:
Should this be `AppendKeyWithMaxTimestamp(&start_with_ts, start, ts_sz);` because we want to create a smallest internal key.

Reply:
Yeah, it's a little bit counterintuitive, I updated and added test for that.